### PR TITLE
Broken GitHub links from gallery examples

### LINKED ISSUE #2108: Broken GH links from gallery examples
When clicking on the GitHub link in the gallery examples, the link to the source code is incorrect. It seems that the GH link is missing the subfolder where the gallery example is placed. See the below example:

Documentation link: https://pvlib-python.readthedocs.io/en/stable/gallery/bifacial/plot_pvfactors_fixed_tilt.html#sphx-glr-gallery-bifacial-plot-pvfactors-fixed-tilt-py
Link to source code: https://github.com/pvlib/pvlib-python/blob/main/docs/examples/plot_pvfactors_fixed_tilt.py

image

### PR DESCRIPTION
**Pull request by IoannisSifnaios**
- [X] Closes #2108
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: docs/sphinx/source/conf.py [hunk 1]
```diff
@@ -445,8 +445,9 @@ def make_github_url(file_name):
         if file_name.split("/")[-1] == "index":
             example_file = "README.rst"
         else:
+            example_folder = file_name.split("/")[-2]
             example_file = file_name.split("/")[-1].replace('.rst', '.py')
-        target_url = URL_BASE + "docs/examples/" + example_file
+        target_url = URL_BASE + "docs/examples/" + example_folder + "/" + example_file  # noqa: E501
```
Comment (kandersolar) [inline docs/sphinx/source/conf.py:450]: One issue: `example_folder` is only defined in the `else` branch here.

Another issue is that the conditional might need to check `file_name.split("/")[-1] == "index.rst"`?  Not 100% sure, but worth trying :)

**Review by kandersolar — COMMENTED** [review]
Looks like the "view on GH" links on these types of pages are still broken: 
- top-level index: https://pvlib-python--2111.org.readthedocs.build/en/2111/gallery/index.html
- sub-level index: https://pvlib-python--2111.org.readthedocs.build/en/2111/gallery/irradiance-transposition/index.html

**Comment by IoannisSifnaios** [comment]
Thanks @kandersolar! I think it is fixed now.

**Review by kandersolar — APPROVED** [review]
LGTM.  No need for a whatsnew entry IMHO.  Thanks @IoannisSifnaios!

**Pull request merged**